Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -162,11 +161,13 @@ AppendPipeline::AppendPipeline(SourceBufferPrivateGStreamer& sourceBufferPrivate
     m_demuxerDataEnteringPadProbeInformation.probeId = gst_pad_add_probe(demuxerPad.get(), GST_PAD_PROBE_TYPE_BUFFER, reinterpret_cast<GstPadProbeCallback>(appendPipelinePadProbeDebugInformation), &m_demuxerDataEnteringPadProbeInformation, nullptr);
 #endif
 
-    if (hasDemuxer) {
+    auto elementClass = makeString(gst_element_get_metadata(m_demux.get(), GST_ELEMENT_METADATA_KLASS));
+    auto classifiers = elementClass.split('/');
+    if (classifiers.contains("Demuxer"_s)) {
         // These signals won't outlive the lifetime of `this`.
-        g_signal_connect(m_demux.get(), "no-more-pads", G_CALLBACK(+[](GstElement*, AppendPipeline* appendPipeline) {
+        g_signal_connect_swapped(m_demux.get(), "no-more-pads", G_CALLBACK(+[](AppendPipeline* appendPipeline) {
```

Comment:
Are you saying we shouldn't use `g_signal_connect_swapped()`? Because it's currently used in many many places :)

Reply:
Then today we learn we have undefined behavior in many many places.

Reply:
I don't think we've received any bug report about this. It may be undefined behavior but so far it has worked for us, unless proven otherwise :)

Reply:
It has worked for us because it just so happens that in x86_64 SysV calling convention, as well as in the calling conventions for ARM 32-bit and ARM 64-bit the first few arguments are passed as registers.

Reply:
After looking more into it (including looking at glib and libffi source code), it seems this is just how glib does things. It's indeed undefined behavior that works only thanks to specifics of popular ABIs, but it seems glib and GTK are okay with it. 

So no point in policing it in WebKit. It's still dangerous for any non-trivial set of arguments.

Reply:
Fun fact from today's talk in the GStreamer Conference 2023 about GStreamer in WebAssembly. This pattern breaks on Emscripten's ABI.